datalad/core/distributed/clone.py

```diff
@@ -0,0 +1,625 @@
+                    message=re_match.group(1) if re_match else "stderr: " + e_stderr,
+                    **result_props)
+                return
+    log_progress(
+        lgr.info,
+        'cloneds',
+        'Completed clone attempts for %s', destds
+    )
+
+    if not destds.is_installed():
+        if len(error_msgs):
+            if all(not e.stdout and not e.stderr for e in error_msgs.values()):
+                # there is nothing we can learn from the actual exception,
+                # the exit code is uninformative, the command is predictable
+                error_msg = "Failed to clone from all attempted sources: %s"
+                error_args = list(error_msgs.keys())
+            else:
+                error_msg = "Failed to clone from any candidate source URL. " \
+                            "Encountered errors per each url were:\n- %s"
+                error_args = '\n- '.join(
+                    '{}\n  {}'.format(url, exc_str(exc))
+                    for url, exc in error_msgs.items()
+                )
+        else:
+            # yoh: Not sure if we ever get here but I felt that there could
+            #      be a case when this might happen and original error would
+            #      not be sufficient to troubleshoot what is going on.
+            error_msg = "Awkward error -- we failed to clone properly. " \
+                        "Although no errors were encountered, target " \
+                        "dataset at %s seems to be not fully installed. " \
+                        "The 'succesful' source was: %s"
+            error_args = (destds.path, source_)
+        yield get_status_dict(
+            status='error',
+            message=(error_msg, error_args),
+            **result_props)
+        return
+
+    yield from _handle_possible_annex_dataset(
+        destds,
+        reckless,
+        description)
+
+    # yield successful clone of the base dataset now, as any possible
+    # subdataset clone down below will not alter the Git-state of the
+    # parent
+    yield get_status_dict(status='ok', **result_props)
+
+
+def _handle_possible_annex_dataset(ds, reckless, description=None):
+    """If ds "knows annex" -- annex init it, set into reckless etc
+
+    Provides additional tune up to a possibly an annex repo, e.g.
+    "enables" reckless mode, sets up description
+    """
+    # in any case check whether we need to annex-init the installed thing:
+    if not knows_annex(ds.path):
+        # not for us
+        return
+
+    # init annex when traces of a remote annex can be detected
+    if reckless:
+        lgr.debug(
+            "Instruct annex to hardlink content in %s from local "
+            "sources, if possible (reckless)", ds.path)
+        ds.config.add(
+            'annex.hardlink', 'true', where='local', reload=True)
+
+    # we have just cloned the repo, so it has 'origin', configure any
+    # reachable origin of origins
+    yield from configure_origins(ds, ds, label=None)
```

Comment:
I feel like that call to `configure_origin` doesn't belong in `_handle_possible_annex_dataset`. As far as I understand what it's for, it's not related to the cloned repo having or not having an annex.
So, move that `yield from` up to line 450 or sth?

Reply:
It only makes sense for annex datasets.

Reply:
Does it? I get, that availability propagation is the purpose, but at the same time - it actually does care for remotes only and one might as well be interested in other branches than `git-annex`. I could see it being confusing to have those remotes automatically available when there is an annex but not when it's plain git. All that functionality relies on, is plain git mechanism. And for PR workflows like ours, I do find such a functionality useful even with plain git repos. The fact that there's no git-annex branch doesn't determine my interest in other remote branches.
How does it "not make sense" for git repo to connect to known other repos with a common history? One might explicitly not want that, but that might be true for annexes as well.

Reply:
It makes sense, if the outcome is desired. But I am not clearly seeing the use case. What is happening is that any local sibling is discovered by retracing the origin chain. One ends up with lots of `origin-?` remotes. For obtaining file content, once doesn't really care what is what and where things actually come from. For merges and such that is not the same. I very much want to know what is what. If we can come up with a proper use case I am not opposed, but without it, I'd say let's keep it limited to annex.

Reply:
I prefer to let this remain an idea until we discover a use case that makes it desirable. Once that happens, it should be easy to shift this functionality in the code to make it happen for GitRepos too, without imposing breakage in other places.